File: src/main/java/com/example/demo/repository/IndicatorRepository.java
package com.example.demo.repository;


import java.sql.ResultSet;
import java.util.List;

import org.springframework.jdbc.core.RowMapper;
import com.example.demo.view.IndicatorView;


public interface IndicatorRepository {

    RowMapper<IndicatorView> INDICATOR_VIEW_ROW_MAPPER = (ResultSet rs, int rowNum) -> {
        IndicatorView indicatorView = new IndicatorView(
                rs.getString("name"),
                rs.getLong("employeeCount"),
                rs.getDouble("averageKpi"));

        return indicatorView;
    };

    List<IndicatorView> findAllWithAvgValue();
}

File: src/main/kotlin/com/example/demo/repository/IndicatorRepository.kt
package com.example.demo.repository

import com.example.demo.view.IndicatorView
import org.springframework.jdbc.core.RowMapper
import java.sql.ResultSet


interface IndicatorRepository {
    fun findAllWithAvgValue(): List<IndicatorView>

    companion object {
        val INDICATOR_VIEW_ROW_MAPPER: RowMapper<IndicatorView> = RowMapper { rs: ResultSet, _: Int ->
            val indicatorView = IndicatorView(
                rs.getString("name"),
                rs.getLong("employeeCount"),
                rs.getDouble("averageKpi")
            )
            indicatorView
        }
    }
}

